src/Microsoft.Azure.SignalR/HubHost/ServiceHubDispatcher.cs

```diff
@@ -83,12 +86,17 @@ private Uri GetServiceUrl(string connectionId)
 
         public async Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, string connectionId, CancellationToken cancellationToken = default)
         {
+            var customHeaders = new Dictionary<string, string>
```

Comment:
Or we can cache the headers instead of creating it every time.

Reply:
Agree. Will update.